lib/common/format.h

```diff
@@ -102,6 +102,30 @@ template<typename CHAR = char> class FormatValidator {
     String,  // char-literal-constant or Hollerith constant
   };
 
+  bool mayHaveLeadingInt(TokenKind desc) {
```

Comment:
This member function should be marked `static` or (better, if possible) `static constexpr`.